### PR TITLE
EN-1281: add support for more JDBC timeout configs

### PR DESCRIPTION
**Pull request by jtoberon**
And strongly type them right into some classes.

A param we want to use cancelSignalTimeout requires 9.4.1209 or higher:
https://jdbc.postgresql.org/documentation/94/connect.html

### CODE REVIEW

**Comment by jtoberon** [comment]
I just want to do the equivalent of java8's ifPresent!

**Comment by aerust** [comment]
If this is related to the soql-servers, why are you doing it in data-coordinator? (not that it shouldn't be done in data-coordinator)

**Comment by jtoberon** [comment]
@aerust because soql-servers use data-coordinator's DataSourceFromConfig class.

File: coordinatorlib/src/main/scala/com/socrata/datacoordinator/common/DataSourceFromConfig.scala [hunk 1]
```diff
@@ -18,6 +18,9 @@ class DataSourceConfig(config: Config, root: String) extends ConfigClass(config,
   val password = getString("password")
   val applicationName = getString("app-name")
   val tcpKeepAlive = optionally(getBoolean("tcp-keep-alive")).getOrElse(false)
+  val loginTimeout = optionally(getInt("login-timeout"))
```
Comment (znep) [inline coordinatorlib/src/main/scala/com/socrata/datacoordinator/common/DataSourceFromConfig.scala]: Use getDuration instead of getInt.

Comment (jtoberon) [inline coordinatorlib/src/main/scala/com/socrata/datacoordinator/common/DataSourceFromConfig.scala]: good idea. pushed.

**Pull request merged**